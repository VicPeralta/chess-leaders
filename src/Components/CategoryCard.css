.category-card {
  width: 150px;
  height: 200px;
  border: 1px solid gray;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.category-card > img {
  width: 100px;
  height: auto;
}
