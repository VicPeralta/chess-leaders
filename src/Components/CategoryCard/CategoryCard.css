.category-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 160px;
  height: 200px;
  border-radius: 8px;
  cursor: pointer;
  margin: 2px auto;
  color: inherit;
  text-decoration: none;
}

.category-title-section {
  display: flex;
  width: 100%;
  justify-content: flex-end;
  align-items: center;
}

.category-card > img {
  width: 100px;
  height: auto;
}

.category-card a {
  text-decoration: none;
  color: white;
}

.category-title {
  color: var(--text-color);
  font-weight: bold;
  font-size: 1.1rem;
  text-decoration: none;
  margin-bottom: 2px;
  text-align: right;
  align-self: flex-end;
}

.average {
  align-self: flex-end;
  font-size: 0.9rem;
}
