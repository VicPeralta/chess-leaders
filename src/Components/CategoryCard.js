import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import daily from '../assets/daily.png';
import daily960 from '../assets/daily960.png';
import liveRapid from '../assets/rapid.png';
import liveBlitz from '../assets/blitz.png';
import liveBullet from '../assets/bullet.png';
import liveBughouse from '../assets/bug.png';
import liveBlitz960 from '../assets/blitz960.png';
import liveThreecheck from '../assets/three.png';
import liveCrazyhouse from '../assets/house.png';
import liveKingofthehill from '../assets/king.png';
import tactics from '../assets/tactics.png';
import rush from '../assets/rush.png';
import battle from '../assets/battle.png';
import './CategoryCard.css';

const icons = {
  daily,
  daily960,
  live_rapid: liveRapid,
  live_blitz: liveBlitz,
  live_bullet: liveBullet,
  live_bughouse: liveBughouse,
  live_blitz960: liveBlitz960,
  live_threecheck: liveThreecheck,
  live_crazyhouse: liveCrazyhouse,
  live_kingofthehill: liveKingofthehill,
  tactics,
  rush,
  battle,
};

const Category = ({ id, name, average }) => (
  <Link to={`/leaderboard/${id}`}>
    <div className="category-card" role="presentation" data-id={id}>
      <p>{name}</p>
      <img src={icons[name]} alt="the category" />
      <p>{average}</p>
    </div>
  </Link>
);

Category.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  average: PropTypes.number.isRequired,
};
export default Category;
