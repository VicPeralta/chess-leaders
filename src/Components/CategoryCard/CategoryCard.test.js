import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import store from '../../Redux/configureStore';
import CategoryCard from './CategoryCard';

describe('CategoryCard Componente test', () => {
  test('Render CategoryCard', () => {
    const element = (
      <Provider store={store}>
        <HashRouter>
          <CategoryCard id={1} name="Blitz" average={2000} />
        </HashRouter>
      </Provider>
    );
    render(element);
    const card = screen.getByText(/Blitz/i);
    expect(card).toBeInTheDocument();
  });
  test('CategoryCard snapshot', () => {
    const element = (
      <Provider store={store}>
        <HashRouter>
          <CategoryCard id={1} name="Blitz" average={2000} />
        </HashRouter>
      </Provider>
    );
    const card = renderer.create(element);
    expect(card).toMatchSnapshot();
  });
});
