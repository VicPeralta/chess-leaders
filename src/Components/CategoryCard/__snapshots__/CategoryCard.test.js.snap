// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Componente test CategoryCard snapshot 1`] = `
<a
  href="#/leaderboard/1"
  onClick={[Function]}
>
  <div
    className="category-card"
    data-id={1}
    role="presentation"
  >
    <div
      className="category-title-section"
    >
      <svg
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
    <img
      alt="the category"
    />
    <p
      className="category-title"
    >
      Blitz
    </p>
    <div
      className="average"
    >
      <span
        className="highlighted"
      >
        avg:  
      </span>
      <span>
        2000
      </span>
    </div>
  </div>
</a>
`;
